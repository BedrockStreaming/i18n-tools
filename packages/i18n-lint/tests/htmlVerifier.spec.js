import HtmlVerifier from '../src/verifiers/htmlVerifier';

jest.unmock('../src/verifiers/htmlVerifier');

const tests = {
  firstTag: 'foo <div> foo',
  lastTag: 'foo </div> foo',
  control: '<div> foo </div>',
  doubleSpace: 'foo  bar',
  doubleSpaceWithNonBreakingSpace: 'foo \u00a0bar',
  doubleSpaceWithHtmlTag: 'foo <foo> bar</foo>',
  doubleSpaceWithHtmlTagClosing: '<foo>foo </foo> bar',
};

describe('htmlVerifier', () => {
  it('should detect html errors', () => {
    expect(HtmlVerifier({ tests }, true)).toMatchSnapshot();
  });
});
