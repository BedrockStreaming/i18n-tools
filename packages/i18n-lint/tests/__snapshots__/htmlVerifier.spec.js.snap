// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmlVerifier should detect html errors 1`] = `
Array [
  Object {
    "error": true,
    "message": "tests: translation for key 'firstTag' has missing html tag 
  foo <div> foo",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'firstTag' has double white space with inner html tag 
  foo <div> foo",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'lastTag' has missing html tag 
  foo </div> foo",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'lastTag' has double white space with inner html tag 
  foo </div> foo",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'doubleSpace' has double white space 
  foo  bar",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'doubleSpaceWithNonBreakingSpace' has double white space 
  foo Â bar",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'doubleSpaceWithHtmlTag' has double white space with inner html tag 
  foo <foo> bar</foo>",
  },
  Object {
    "error": true,
    "message": "tests: translation for key 'doubleSpaceWithHtmlTagClosing' has double white space with inner html tag 
  <foo>foo </foo> bar",
  },
]
`;
